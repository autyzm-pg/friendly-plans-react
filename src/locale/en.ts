const translations = {
  planActivity: {
    planNamePlaceholder: 'Type plan name',
    shuffleTasks: 'Shuffle tasks',
  },
  common: {
    cancel: 'Cancel',
    ok: 'OK',
    yes: 'Yes',
    email: 'Email address',
    name: 'Your Name',
    password: 'Password',
    error: 'Error',
    success: 'Success',
    unknownError: 'Unknown error occured',
    addImage: 'Add Image',
  },
  validation: {
    email: 'Must be a valid email',
    required: 'This field is required',
    passwordLength: 'Password is too short',
    acceptRequired: 'You need to accept our Terms of Use',
    duplicatedPlan: 'Plan name already exists',
    planNameRequired: 'Plan name is required',
  },
  notifications: {
    channelName: 'Notifications',
    channelDescription: 'Get urgent notifications about the service',
  },
  signIn: {
    signIn: 'Sign In',
    signInButton: 'Sign In',
    signInAsAnonymous: 'Use the App as Guest',
    signUpTip: "don't have an account?",
    anonymousTip: 'you can also',
    forgotPassword: 'forgot password?',
  },
  signUp: {
    signUp: 'Sign Up',
    signUpButton: 'Sign up',
    accept: 'I accept ',
    termsOfUse: 'Terms of Use',
  },
  resetPassword: {
    resetPassword: 'Reset password',
    guide1: 'Provide you email in the field below. We will send you link to reset your password.',
    guide2: 'After setting up new password, sign in with newly set password at the login screen.',
    resetPasswordButtonLabel: 'Reset Password',
    userNotFound: 'No user found for provided email. Make sure to provide email you registered.',
    resetPasswordSuccess:
      'Email with password reset link has been sent. After setting up new password, sign in with newly set password at the login screen.',
  },
  content: {
    content: 'Content',
    contentList: 'Content List',
  },
  settings: {
    settings: 'Settings',
    signOutTitle: 'Sign out',
    signOutSubtitle: 'After you sign out, you will need to sign in again.',
    signOutAction: 'Sign out',
    signOutDialogTitle: 'Sign out',
    signOutDialogDescription: 'Are you sure you want to sign out?',
  },
  sidebar: {
    addTask: 'Add SimpleTask',
    addStudent: 'Add Student',
    takeAPicture: 'Take a Picture',
    recordSound: 'Record Sound',
  },
  studentList: {
    createStudent: 'Add a Student',
    screenTitle: 'Select student',
    search: 'Search',
    dashboard: 'Student dashboard',
    studentNamePlaceholder: 'Enter Student name...',
    removeStudentTitle: 'Confirm',
    removeStudentDescription:
      'Are you sure you want to remove {{name}} from your student list?\n This action cannot be undone.',
  },
  planList: {
    viewTitle: 'All Plans',
    createPlan: 'Add new plan',
    copyPlan: 'Copy existing plan',
    conjunction: 'lub',
    planNamePlaceholder: 'Muzykowanie',
  },
  planItemActivity: {
    viewTitleTask: 'Task',
    viewTitleInteraction: 'Interaction',
    viewTitleBreak: 'Break',
    taskNamePlaceholder: 'Type task name',
    timerButton: 'Set up timer',
    taskNameForChild: 'Type task name for child',
    addImage: 'Add image',
    imageActionTakePhoto: 'Take a photo',
    imageActionLibrary: 'Browse from library',
    imageActionBrowse: 'Browse from device',
    imageLibraryTitle: 'Image Library',
  },
  updatePlan: {
    screenTitle: '{{studentName}} - Update a plan',
    removePlanTitle: 'Confirm',
    removePlanDescription:
      'Are you sure you want to plan "{{name}}" from student plan list?\n This action cannot be undone.',
    addBreak: 'Add break',
    addInteraction: 'Add interaction',
    addTask: 'Add task',
    planItemNamePlaceholder: 'Enter Plan Item name...',
  },
  studentSettings: {
    studentName: "Student's name",
    taskView: "Task's view",
    soundSettings: 'Sound settings',
    settingsTitle: "Student's settings",
    createStudentTitle: 'Create a Student',
    textSettingsSizeS: 'Font size: S',
    textSettingsSizeM: 'Font size: M',
    textSettingsSizeL: 'Font size: L',
    textSettingsSizeXL: 'Font size: XL',
    largeImageSlide: 'Large image as a slide',
    imageWithTextSlide: 'Image with text label as a slide',
    textSlide: 'Just text as a slide',
    imageWithTextList: 'image with text label as a list',
    textList: 'Just text label as a list',
    uppercase: 'Uppercase letters',
    blockSwipe: 'Block swipe',
    alarmSound: 'Alarm sound',
    planCardPlacehorder: "Let's play paper, rock, scissors",
    studentNamePlaceholder: 'Enter student name',
    removeStudent: 'Remove student',
    createStudent: 'Create student',
  },
  runPlan: {
    next: 'Next',
    wait: 'Wait...',
  },
  taskTable: {
    number: '#',
    name: 'Name',
    type: 'Type',
    section: 'Parts',
    time: 'Time',
    delete: 'Delete',
    edit: 'Edit',
  },
};

export default translations;
