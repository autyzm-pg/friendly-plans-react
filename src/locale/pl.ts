const translations = {
  planActivity: {
    planNamePlaceholder: 'Wpisz nazwę planu',
    shuffleTasks: 'Tasuj zadania',
  },
  common: {
    cancel: 'Anuluj',
  },
  signIn: {
    signIn: 'Zaloguj się',
    signInAsAnonymous: 'Zaloguj się anonimowo',
  },
  content: {
    content: 'Treść',
    contentList: 'Treści',
  },
  settings: {
    settings: 'Ustawienia',
    signOutTitle: 'Wyloguj się',
    signOutSubtitle: 'Po wylogowaniu się, musisz zalogować się ponownie',
    signOutAction: 'Wyloguj',
    signOutDialogTitle: 'Wyloguj się',
    signOutDialogDescription: 'Czy na pewno chcesz się wylogować?',
  },
  planList: {
    viewTitle: 'Wszystkie plany',
    createPlan: 'DODAJ NOWY PLAN',
    copyPlan: 'SKOPIUJ ISTNIEJĄCY PLAN',
    conjunction: 'lub',
    planNamePlaceholder: 'Muzykowanie',
  },
  planItemActivity: {
    viewTitleTask: 'Zadanie',
    viewTitleInteraction: 'Interakcja',
    viewTitleBreak: 'Przerwa',
    taskNamePlaceholder: 'Wpisz nazwę zadania',
    timerButton: 'Ustaw timer',
    taskNameForChild: 'Nazwij dla dziecka',
    addImage: 'Dodaj obrazek',
    imageActionTakePhoto: 'Zrób zdjęcie',
    imageActionLibrary: 'Wybierz z biblioteki',
    imageActionBrowse: 'Dodaj z urządzenia',
    imageLibraryTitle: 'Biblioteka zdjęć',
  },
  updatePlan: {
    addBreak: 'Dodaj przerwę',
    addInteraction: 'Dodaj interakcję',
    addTask: 'Dodaj zadanie',
  },
  studentSettings: {
    studentName: 'Imię podopiecznego',
    taskView: 'Widok zadań',
    soundSettings: 'Ustawienia dźwięku',
    settingsTitle: 'Ustawienia podopiecznego',
    createStudentTitle: 'Stwórz Studenta',
    textSettingsSizeS: 'Rozmiar liter: S',
    textSettingsSizeM: 'Rozmiar liter: M',
    textSettingsSizeL: 'Rozmiar liter: L',
    textSettingsSizeXL: 'Rozmiar liter: XL',
    largeImageSlide: 'Duży obrazek jako slajd',
    imageWithTextSlide: 'Obrazek z tekstem jako slajd',
    textSlide: 'Tylko tekst jako slajd',
    imageWithTextList: 'Obrazek z tekstem jako lista',
    textList: 'Tylko tekst jako lista',
    uppercase: 'Wielkie litery',
    blockSwipe: "Blokada swipe'u",
    alarmSound: 'Dźwięk alarmu',
    planCardPlacehorder: 'Zagraj w papier, kamień, nożyce',
    studentNamePlaceholder: 'Wpisz imię podopiecznego',
    removeStudent: 'Usuń studenta',
    createStudent: 'Dodaj Studenta',
  },
  studentsList: {
    screenTitle: 'Wybór podopiecznego',
    search: 'Wyszukaj',
    dashboard: 'Studenci',
    createStudent: 'Stwórz studenta',
  },
  runPlan: {
    next: 'Dalej',
    wait: 'Czekaj...',
  },
  taskTable: {
    number: '#',
    name: 'Nazwa',
    type: 'Typ',
    section: 'Części',
    time: 'Czas',
    delete: 'Usuń',
    edit: 'Edycja',
  },
  validation: {
    duplicatedPlan: 'Plan o podanej nazwie już istnieje',
    planNameRequired: 'Nazwa planu jest wymagana',
  },
};

export default translations;
