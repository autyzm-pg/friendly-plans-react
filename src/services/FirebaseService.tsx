import firebase from 'react-native-firebase';
import { Image } from 'react-native-image-crop-picker';

export class FirebaseService {
  /* Upload image to Firebase Storage and return public URL to download it. */
  static uploadUserImage = async (image: string): Promise<string> => {
    const user = firebase.auth().currentUser!;
    const imageRef = firebase
      .storage()
      .ref('users')
      .child(user.uid);
    const snapshot = await imageRef.putFile(image);
    return snapshot.downloadURL as string;
  };

  static updateUserImage = async (image: string): Promise<void> => {
    const photoURL = await FirebaseService.uploadUserImage(image);
    return firebase.auth().currentUser!.updateProfile({ photoURL });
  };
}
