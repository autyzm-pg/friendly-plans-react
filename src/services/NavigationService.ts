import { NavigationActions, NavigationContainerComponent, NavigationState } from 'react-navigation';

import { DialogProps } from 'models';

interface ScreenProps extends Partial<DialogProps> {
  [property: string]: any;
}

export default class NavigationService {
  navigator?: NavigationContainerComponent;

  setTopLevelNavigator = (navigatorRef: NavigationContainerComponent) => {
    this.navigator = navigatorRef;
  };

  navigate = (routeName: string, params?: ScreenProps) => {
    this.navigator!.dispatch(
      NavigationActions.navigate({
        routeName,
        params,
      }),
    );
  };

  goBack = () => {
    this.navigator!.dispatch(NavigationActions.back());
  };

  getActiveRouteName = (navigationState: NavigationState): string => {
    if (!navigationState) {
      return '';
    }
    const route = navigationState.routes[navigationState.index];
    // dive into nested navigators
    if (route.routes) {
      return this.getActiveRouteName(route as NavigationState);
    }
    return route.routeName;
  };
}
