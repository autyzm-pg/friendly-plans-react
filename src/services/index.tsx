import _AnalyticsService from './AnalyticsService';
import _CrashlyticsService from './CrashlyticsService';
import _NavigationService from './NavigationService';

export const NavigationService = new _NavigationService();
export const AnalyticsService = new _AnalyticsService(NavigationService);
export const CrashlyticsService = new _CrashlyticsService();
