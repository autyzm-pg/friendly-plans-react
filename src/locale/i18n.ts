import i18n, { LanguageDetectorModule } from 'i18next';
import { initReactI18next } from 'react-i18next';
import { getLocales } from 'react-native-localize';

import en from './en';
import pl from './pl';

const languageDetector: LanguageDetectorModule = {
  type: 'languageDetector',
  init(services, detectorOptions, i18nextOptions) {
    /* use services and options */
  },
  detect() {
    return getLocales()[0].languageCode;
  },
  cacheUserLanguage(lng) {
    /* cache language */
  },
};

i18n
  .use(languageDetector)
  .use(initReactI18next)
  .init({
    fallbackLng: 'en',
    resources: {
      en,
      pl,
    },
    ns: ['common', 'signIn', 'signUp'],
    defaultNS: 'common',
    debug: __DEV__,
    interpolation: {
      escapeValue: false,
    },
  });

export default i18n;
