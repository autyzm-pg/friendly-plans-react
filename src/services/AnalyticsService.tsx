import firebase from 'react-native-firebase';
import { NavigationState } from 'react-navigation';

import { User } from 'models';
import NavigationService from './NavigationService';

export default class AnalyticsService {
  navigationService: NavigationService;
  user: User | null;
  screenName?: string;

  constructor(navigationService: NavigationService) {
    firebase.analytics().setAnalyticsCollectionEnabled(!__DEV__);
    this.navigationService = navigationService;
    this.user = null;
  }

  setCurrentScreen = (prevState: NavigationState, currentState: NavigationState) => {
    const currentScreen = this.navigationService.getActiveRouteName(currentState);
    const prevScreen = this.navigationService.getActiveRouteName(prevState);

    if (prevScreen !== currentScreen) {
      firebase.analytics().setCurrentScreen(currentScreen, currentScreen);
      this.screenName = currentScreen;
    }
  };

  setCurrentUser = (user: User | null) => {
    this.user = user;
    firebase.analytics().setUserId(this.user ? this.user!.id.toString() : null);
  };
}
