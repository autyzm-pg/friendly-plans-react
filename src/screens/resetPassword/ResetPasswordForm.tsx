import { Formik, FormikProps } from 'formik';
import React from 'react';
import { Alert, StyleSheet, View } from 'react-native';
import firebase from 'react-native-firebase';
import * as Yup from 'yup';

import { Button, InputItem } from 'components';
import { i18n } from 'locale';
import { NavigationService } from '../../services';

const validationSchema = Yup.object().shape({
  email: Yup.string()
    .email(i18n.t('validation:email'))
    .required(i18n.t('validation:required')),
});

const initialValues = {
  email: '',
};

interface ResetPasswordFormData {
  email: string;
}

interface State {
  loading: boolean;
}

export class ResetPasswordForm extends React.PureComponent<{}, State> {
  state = {
    loading: false,
  };

  onSubmit = async (values: ResetPasswordFormData) => {
    try {
      await firebase.auth().sendPasswordResetEmail(values.email);
      Alert.alert(i18n.t('common:success'), i18n.t('resetPassword:resetPasswordSuccess'));
      NavigationService.navigate('SignIn');
    } catch (error) {
      if (error.code === 'auth/user-not-found') {
        Alert.alert(i18n.t('common:error'), i18n.t('resetPassword:userNotFound'));
      } else {
        Alert.alert(i18n.t('common:error'), i18n.t('common:unknownError'));
      }
    }
  };

  render() {
    return (
      <Formik
        initialValues={initialValues}
        onSubmit={this.onSubmit}
        validationSchema={validationSchema}
        render={this.renderForm}
      />
    );
  }

  renderForm = (props: FormikProps<ResetPasswordFormData>) => {
    const { handleChange, handleBlur, values, handleSubmit, errors, touched } = props;
    return (
      <View style={styles.container}>
        <InputItem
          onChangeText={handleChange('email')}
          onBlur={handleBlur('email')}
          value={values.email}
          error={errors.email}
          touched={touched.email}
          label={i18n.t('common:email')}
          autoCapitalize="none"
          keyboardType="email-address"
          textContentType="emailAddress"
          blurOnSubmit
        />
        <Button
          onPress={handleSubmit}
          title={i18n.t('resetPassword:resetPasswordButtonLabel')}
          containerStyle={styles.button}
        />
      </View>
    );
  };
}

const styles = StyleSheet.create({
  container: {
    marginTop: 40,
  },
  button: {
    marginTop: 8,
  },
});
