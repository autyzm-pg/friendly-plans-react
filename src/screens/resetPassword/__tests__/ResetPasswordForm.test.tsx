import { shallow } from 'enzyme';
import React from 'react';
import { Alert } from 'react-native';
import firebase from 'react-native-firebase';

import { i18n } from 'locale';
import { NavigationService } from 'services';
import { ResetPasswordForm } from '../ResetPasswordForm';

describe('ResetPasswordForm', () => {
  const formProps = {
    handleChange: jest.fn(),
    handleBlur: jest.fn(),
    values: {
      email: 'email@domain.com',
    },
    errors: {
      email: null,
    },
    touched: {
      email: true,
    },
    handleSubmit: jest.fn(),
  } as any;

  it('should match snapshot', () => {
    const wrapper = shallow(<ResetPasswordForm />);
    expect(wrapper).toMatchSnapshot();
  });

  it('form should match snapshot', () => {
    const wrapper = shallow(<ResetPasswordForm />);
    const instance = wrapper.instance() as ResetPasswordForm;
    const form = shallow(instance.renderForm(formProps));
    expect(form).toMatchSnapshot();
  });

  it('should show success message and navigate to sign in screen on successful password reset', async () => {
    const wrapper = shallow(<ResetPasswordForm />);
    const instance = wrapper.instance() as ResetPasswordForm;
    const navigationSpy = jest.spyOn(NavigationService, 'navigate').mockImplementationOnce(jest.fn());
    const alertSpy = jest.spyOn(Alert, 'alert');
    await instance.onSubmit(formProps.values);
    expect(navigationSpy).toBeCalledWith('SignIn');
    expect(alertSpy).toBeCalledWith(i18n.t('common:success'), i18n.t('resetPassword:resetPasswordSuccess'));
  });

  it('should handle user not found errors', async () => {
    const wrapper = shallow(<ResetPasswordForm />);
    const instance = wrapper.instance() as ResetPasswordForm;
    const spy = jest.spyOn(Alert, 'alert');
    // @ts-ignore
    firebase.auth = jest.fn().mockImplementationOnce(() => ({
      sendPasswordResetEmail: jest.fn(() => Promise.reject({ code: 'auth/user-not-found' })),
    }));
    await instance.onSubmit(formProps.values);
    expect(spy).toBeCalledWith(i18n.t('common:error'), i18n.t('resetPassword:userNotFound'));
  });

  it('should handle unknown errors', async () => {
    const wrapper = shallow(<ResetPasswordForm />);
    const instance = wrapper.instance() as ResetPasswordForm;
    const spy = jest.spyOn(Alert, 'alert');
    // @ts-ignore
    firebase.auth = jest.fn().mockImplementationOnce(() => ({
      sendPasswordResetEmail: jest.fn(() => Promise.reject({ code: 'unknown code' })),
    }));
    await instance.onSubmit(formProps.values);
    expect(spy).toBeCalledWith(i18n.t('common:error'), i18n.t('common:unknownError'));
  });
});
