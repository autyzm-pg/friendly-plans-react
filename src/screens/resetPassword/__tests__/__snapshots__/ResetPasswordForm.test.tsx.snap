// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResetPasswordForm form should match snapshot 1`] = `
<View
  style={
    Object {
      "marginTop": 40,
    }
  }
>
  <InputItem
    autoCapitalize="none"
    blurOnSubmit={true}
    error={null}
    keyboardType="email-address"
    label="Email address"
    textContentType="emailAddress"
    touched={true}
    value="email@domain.com"
  />
  <Component
    backgroundColor="#6654EB"
    containerStyle={
      Object {
        "marginTop": 8,
      }
    }
    onPress={[MockFunction]}
    title="Reset Password"
  />
</View>
`;

exports[`ResetPasswordForm should match snapshot 1`] = `
<Formik
  enableReinitialize={false}
  initialValues={
    Object {
      "email": "",
    }
  }
  isInitialValid={false}
  onSubmit={[Function]}
  render={[Function]}
  validateOnBlur={true}
  validateOnChange={true}
  validationSchema={
    ObjectSchema {
      "_blacklist": RefSet {
        "list": Set {},
        "refs": Map {},
      },
      "_conditions": Array [],
      "_defaultDefault": [Function],
      "_deps": Array [],
      "_excludedEdges": Array [],
      "_exclusive": Object {},
      "_mutate": undefined,
      "_nodes": Array [
        "email",
      ],
      "_options": Object {
        "abortEarly": true,
        "recursive": true,
      },
      "_type": "object",
      "_typeError": [Function],
      "_whitelist": RefSet {
        "list": Set {},
        "refs": Map {},
      },
      "fields": Object {
        "email": StringSchema {
          "_blacklist": RefSet {
            "list": Set {},
            "refs": Map {},
          },
          "_conditions": Array [],
          "_deps": Array [],
          "_exclusive": Object {
            "required": true,
            "undefined": false,
          },
          "_mutate": undefined,
          "_options": Object {
            "abortEarly": true,
            "recursive": true,
          },
          "_type": "string",
          "_typeError": [Function],
          "_whitelist": RefSet {
            "list": Set {},
            "refs": Map {},
          },
          "tests": Array [
            [Function],
            [Function],
          ],
          "transforms": Array [
            [Function],
          ],
        },
      },
      "tests": Array [],
      "transforms": Array [
        [Function],
      ],
    }
  }
/>
`;
